### ISSUE TITLE
broken link with badger::badge_codecov() - lack codecov token?  

### ISSUE THREAD

**Issue by fontikar** [GuangchuangYu/badger#44]
Hi @GuangchuangYu! Firstly, thank you for making such a convenient package! I use it regularly for package devs!

I am trying to establish a workflow for setting up:
- codecov for a simple, public [R package]((https://github.com/fontikar/ohwhaley/tree/gha))  using `use_coverage(type ="codecov")`
- GitHub Actions CI workflow `test-coverage` using `usethis::use_github_action("test-coverage")`
- a badge for code coverage using `badger::badge_codecov(branch="gha")`

Unfortunately when I use the output for `badger::badge_codecov(branch="gha")` e.g. [d9e2987](https://github.com/fontikar/ohwhaley/tree/d9e2987df893091db4df70273cf52fd1f7509b50) in the README, the badge"shows "unknown: Knitting locally shows the same thing. However navigating to: https://codecov.io/gh/fontikar/ohwhaley/branch/gha/graph/badge.svg, gives me a perfect 100% badge.

I then tried using the codecov badge found in the settings section of the repo which contains my codecov token `https://codecov.io/gh/fontikar/ohwhaley/branch/gha/graph/badge.svg?token=XXXXXXXX` and it was successful e.g. [701020d](https://github.com/fontikar/ohwhaley/tree/701020d1366da8e53999ca9462fed13ce1c0c0e0)

Am I missing a step? I tried adding the codecov token in my codecov.yaml but that didn't work either e.g. [7cd93db](https://github.com/fontikar/ohwhaley/tree/7cd93dbcc51deb7c6ff50465873ad6fe7c529beb)

Your advice would be greatly appreciated!

**Comment by bschilder** [comment]
Hi @GuangchuangYu @wleoncio @rossellhayes, was wondering if any of you (or any of the other Contributors) could speak to this?

Seems to even be an issue on the README of `badger`:
![Screenshot 2023-01-12 at 13 20 12](https://user-images.githubusercontent.com/34280215/212077205-3088119c-3571-4a22-a87d-54156493f025.png)

**Comment by wleoncio** [comment]
What happens if you hardcode both `ref` and `branch`? Usign the example above, I only get a valid output if I specify both:

``` r
badger::badge_codecov("rcannood/princurve", branch = "master")
#> [1] "[![](https://app.codecov.io/gh/rcannood/princurve/branch/master/graph/badge.svg)](https://app.codecov.io/gh/rcannood/princurve)"
```

<sup>Created on 2023-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by bschilder** [comment]
Thanks for the reply @wleoncio 

# Use case 1: `princurve`

In the case of `princurve`, I think you get the wrong output because you may not have it installed localled. Thus the necessary information can't be pulled from the DESCRIPTION file, and some assumptions are made about the URL (ie omitting the owner's name `rcannood`). Though leaving ref as the default yield the same result:

```R
out1=badger::badge_codecov("rcannood/princurve", branch = "master")
out2=badger::badge_codecov("rcannood/princurve")

out1==out2
# TRUE
```

However, when I add the output you generated to a README.Rmd file and try to render it as markdown, it yields a broken link:

### Raw markdown text

```
[![](https://app.codecov.io/gh/rcannood/princurve/branch/master/graph/badge.svg)](https://app.codecov.io/gh/rcannood/princurve)
```

### Markdown rendering
![Screenshot 2023-01-12 at 13 58 52](https://user-images.githubusercontent.com/34280215/212086116-5f810c38-dda3-4bb2-8f96-70d1c815c38f.png)
 
This is odd since navigating to the svg link does indeed show a valid badge:
https://app.codecov.io/gh/rcannood/princurve/branch/master/graph/badge.svg

However, it appears this repo has been deactivated on CodeCov:
https://app.codecov.io/gh/rcannood/princurve

So maybe the issue in this case is the repo being used as example? Though in theory, as long as the badge icon exists, it should still render (it'll just lead to an invalid codecov page).

# Use case 2: `rworkflows`

Let's try with my package instead, `rworkflows`.
It is currently activated on CodeCov:
https://app.codecov.io/gh/neurogenomics/rworkflows


This yield the wrong link, as it omits the owner name ("neurogenomics"):
```R
badger::badge_codecov(ref="rworkflows")
#  "[![](https://app.codecov.io/gh/rworkflows/branch/master/graph/badge.svg)](https://app.codecov.io/gh/rworkflows)"
```
This yield the correct link, including the owner name:
```R
badger::badge_codecov()
```
As does specifying the owner/repo explicitly:

```R
badger::badge_codecov(ref="neurogenomics/rworkflows)
#  "[![](https://app.codecov.io/gh/neurogenomics/rworkflows/branch/master/graph/badge.svg)](https://app.codecov.io/gh/neurogenomics/rworkflows)"
```
The badge icon does indeed exist:
![Screenshot 2023-01-12 at 14 10 09](https://user-images.githubusercontent.com/34280215/212088633-d27ad264-9664-48fd-979a-5bfd1e6d48d5.png)

As does the link:
![Screenshot 2023-01-12 at 14 10 59](https://user-images.githubusercontent.com/34280215/212088743-d05a2ed3-edc6-49a9-a847-81a6945994aa.png)

However, just like the `princurve` output, the badge does not render in markdown.
This also appears to be an issue in html pages, as in my documentation website (notice the broken links in between other badges).
https://neurogenomics.github.io/rworkflows/

Is this a simple markdown syntax issue with the output of `badger::badge_codecov`? If so, I can't seem to figure it out.

**Comment by bschilder** [comment]
Ah, excellent! Thanks so much for this @wleoncio 

So it seems this is just a matter of badger updating the domain URL. Would you (or one of the other developers) possibly be able to make a push to GitHub and CRAN adding this?

**Comment by wleoncio** [comment]
Sure think, I'll issue the PR right away. As for the CRAN release, we're dependent on the maintainer (@GuangchuangYu).

**Comment by bschilder** [comment]
Awesome, thanks @wleoncio !

In the meantime, I'm wrapping a fix around the function when using it within my packages:
```R
gsub("app.codecov.io","codecov.io", ## fix domain name
                         badger::badge_codecov()
                         )
```

**Comment by wleoncio** [comment]
@fontikar, can you confirm that the proposal on PR #46 would solve this issue?

**Comment by fontikar** [comment]
Hi folks! Thanks for everyones efforts and investigations! Its been a while since I looked at this but the issue seemed to resolve itself when I just tried it again

Using `badger::badge_codecov()` and pasting its output in my README seemed to work well: https://github.com/fontikar/ohwhaley 

Recently I taught a `[R packages workshop ](https://fontikar.github.io/DIY_Rpkg_GHA/#Covering_your_bases_%F0%9F%AA%A4_with_codecov)`which involved setting up CodeCov and adding a badge. The participant's badge all initially didn't render, some says unknown, but after granting third party access and waiting a little while, everyone badge was ok? 

Sorry I don't have any further issues and can't provide anything reproducible either at this stage so happy for the PR to go ahead!
